### DISCUSSION TITLE
How to configure renovate to find one dependency in 2 different files ?

### DISCUSSION THREAD

**Discussion by ZeBidule** [renovatebot/renovate#25426] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.33.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to make renovate finding one dependency in 2 different files, for example :

`.tool-repo` file
```
REPO=helm/helm
```

`.tool-versions` file
```
ADDON_VERSION=3.13.0
```

And I want to renovate to create one PR that propose to update `.tool-versions` file with `ADDON_VERSION=3.13.1` because he found this is the last release of the `helm/helm` github repository. 

### What you tried so far.

The closest configuration I found is :
```
{
  "extends": [
    "config:recommended",
    ":disableDependencyDashboard",
    ":disableRateLimiting"
  ],
  "labels": [
    "dependencies"
  ],
  "assigneesFromCodeOwners": true,
  "lockFileMaintenance": {
    "enabled": true
  },
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": ["^\\.tools-versions$","^\\.tools-repo$"],
      "matchStringsStrategy": "combination",
      "matchStrings": [
        "ADDON_VERSION=(?<currentValue>.*)",
        "REPO=(?<depName>.*)"
      ],
      "datasourceTemplate": "github-releases"
    }
  ]
}
```

This configuration works when each file is containing both REPO and ADDON_VERSION information but not in the usecase above.

### Relevant debug logs

<details><summary> non working Logs extract (1 dependency in 2 files)</summary>

```
DEBUG: Matched 2 file(s) for manager regex: .tools-repo, .tools-versions (repository=2182)
DEBUG: manager extract durations (ms) (repository=2182)
       "managers": {"regex": 12}
DEBUG: Found 0 package file(s) (repository=2182)
 INFO: Dependency extraction complete (repository=2182, baseBranch=main)
       "stats": {"managers": {}, "total": {"fileCount": 0, "depCount": 0}}
```

</details>

<details><summary> working Logs extract (1 dependency in each file)</summary>

example :
`.tool-repo` file
```
REPO=helm/helm
ADDON_VERSION=3.12.0
```

`.tool-versions` file
```
REPO=helm/helm
ADDON_VERSION=3.13.0
```

```
DEBUG: Matched 2 file(s) for manager regex: .tools-repo, .tools-versions (repository=2182)
DEBUG: manager extract durations (ms) (repository=2182)
       "managers": {"regex": 4}
DEBUG: Found regex package files (repository=2182)
DEBUG: Found 2 package file(s) (repository=2182)
 INFO: Dependency extraction complete (repository=2182, baseBranch=main)
       "stats": {
         "managers": {"regex": {"fileCount": 2, "depCount": 2}},
         "total": {"fileCount": 2, "depCount": 2}
       }
```

</details>

**Comment by rarkins** [comment]
Not possible. Does the helm/helm string change? You can hard code that into the regex manager using depNameTemplate

**Reply by ZeBidule** [reply]
Thanks for your quick answer, yes the repo name change in my real usecase because I run helm embedded in kustomize with the chart version that is coming from an environment variable :

`.version`
```
export ADDON_VERSION="v23.0.1"
```

`kustomization.yaml`
```
helmCharts:
- name: traefik
  releaseName: traefik
  repo: https://helm.traefik.io/traefik
  version: ${ADDON_VERSION}
```

**Reply by rarkins** [reply, marked as answer]
A custom manager cannot stitch together a dependency from two files

**Comment by ZeBidule** [comment]
Thanks, I will use the renovate annotation system in my `.version` file :
```
# renovate: datasource=github-releases depName=crossplane/crossplane
export ADDON_VERSION="v1.12.1"
```
